### PR TITLE
`az group export` fix output on failure

### LINKED ISSUE #3889: az group export extracts arguably the wrong error message from result and excludes the whole template if anything fails to export
### Description
There are plenty of resource types that currently fail on export (service issue). From the portal, you will get everything that was possible to export + list of error messages for things that couldn't be exported. From the CLI, you just get the first sub-error and nothing else if there is one or more errors.

For example, when running with verbose, you will see the following data in the log - but the error message is only "Could not get resources of the type 'Microsoft.Network/loadBalancers'. Resources of this type will not be exported" and the partial valid template is not output at all...

<partial valid template here>
,"error":{"code":"ExportTemplateCompletedWithErrors","message":"Export template operation completed with errors. Some resources were not exported. Please see details for more information.","details":[{"code":"ExportTemplateProviderError","target":"Microsoft.Network/loadBalancers","message":"Could not get resources of the type \'Microsoft.Network/loadBalancers\'. Resources of this type will not be exported."},{"code":"ExportTemplateProviderError","target":"Microsoft.Network/networkInterfaces","message":"Could not get resources of the type \'Microsoft.Network/networkInterfaces\'. Resources of this type will not be exported."},{"code":"ExportTemplateProviderError","target":"Microsoft.Network/networkSecurityGroups","message":"Could not get resources of the type \'Microsoft.Network/networkSecurityGroups\'. Resources of this type will not be exported."},{"code":"ExportTemplateProviderError","target":"Microsoft.Network/publicIPAddresses","message":"Could not get resources of the type \'Microsoft.Network/publicIPAddresses\'. Resources of this type will not be exported."},{"code":"ExportTemplateProviderError","target":"Microsoft.Network/virtualNetworks","message":"Could not get resources of the type \'Microsoft.Network/virtualNetworks\'. Resources of this type will not be exported."},{"code":"ExportTemplateProviderError","target":"Microsoft.Web/sites/premieraddons","message":"Could not get resources of the type \'Microsoft.Web/sites/premieraddons\'. Resources of this type will not be exported."},{"code":"ExportTemplateProviderError","target":"Microsoft.Web/sites/sourcecontrols","message":"Could not get resources of the type \'Microsoft.Web/sites/sourcecontrols\'. Resources of this type will not be exported."}]}}'

---

### Environment summary

**Install Method:** How did you install the CLI? (e.g. pip, interactive script, apt-get, Docker, MSI, nightly)  
Pip

**CLI Version:** What version of the CLI and modules are installed? (Use `az --version`)  
Latest

**OS Version:** What OS and version are you using?  
Windows

**Shell Type:** What shell are you using? (e.g. bash, cmd.exe, Bash on Windows)  
Bash

### PR DESCRIPTION
**Pull request by tjprescott**
Fixes #3889.

The command will now output the template object. If there are failures, it will output a warning about the failures, following by an error message for each failure.  Example:

```
{
  <partial ARM template>
}

Export template operation completed with errors. Some resources were not exported. Please see details for more information.
Could not get resources of the type 'Microsoft.Sql/servers/connectionPolicies'. Resources of this type will not be exported.
Could not get resources of the type 'Microsoft.Sql/servers/disasterRecoveryConfiguration'. Resources of this type will not be exported.
Could not get resources of the type 'Microsoft.Sql/servers/databases/advisors'. Resources of this type will not be exported.
Could not get resources of the type 'Microsoft.Sql/servers/databases/auditingPolicies'. Resources of this type will not be exported.
Could not get resources of the type 'Microsoft.Sql/servers/databases/backupLongTermRetentionPolicies'. Resources of this type will not be exported.
Could not get resources of the type 'Microsoft.Sql/servers/databases/dataMaskingPolicies'. Resources of this type will not be exported.
Could not get resources of the type 'Microsoft.Sql/servers/databases/extensions'. Resources of this type will not be exported.
Could not get resources of the type 'Microsoft.Sql/servers/databases/geoBackupPolicies'. Resources of this type will not be exported.
Could not get resources of the type 'Microsoft.Sql/servers/databases/securityAlertPolicies'. Resources of this type will not be exported.
Could not get resources of the type 'Microsoft.Sql/servers/databases/transparentDataEncryption'. Resources of this type will not be exported.
Could not get resources of the type 'Microsoft.Sql/servers/databases/auditingSettings'. Resources of this type will not be exported.
Could not get resources of the type 'Microsoft.Sql/servers/databases/syncGroups'. Resources of this type will not be exported.
```

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [X] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [N/A] Each command and parameter has a meaningful description.
- [N/A] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5555
This is an **experimental preview** for @microsoft users.

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py [hunk 1]
```diff
@@ -470,16 +470,15 @@ def export_group_as_template(
...
     # On error, server still returns 200, with details in the error attribute
     if result.error:
         error = result.error
-        if (hasattr(error, 'details') and error.details and
-                hasattr(error.details[0], 'message')):
-            error = error.details[0].message
-        raise CLIError(error)
-
-    print(json.dumps(result.template, indent=2))
+        logger.warning(result.error.message)
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py]: Hope it helps... So you have verified that `result.error` indeed have the `message` field, right?

Comment (tjprescott) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py]: I can add that validation back in. I'm not sure how non-uniform the errors might be from arm. Thanks for the backup!

Comment (tjprescott) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py]: Done

**Pull request merged**